### PR TITLE
[M6 Orders With Coupons] Expandable product cards in order creation - part 1

### PR DESCRIPTION
**Pull request by samiuelson**
## Expandable product cards in order creation - part 1
This is part 1 of the following PRs:
* **Part 1: https://github.com/woocommerce/woocommerce-android/pull/9851**
* Part 2: https://github.com/woocommerce/woocommerce-android/pull/9852
* Part 3: https://github.com/woocommerce/woocommerce-android/pull/10018

### Description

This PR updates `ProductUIModel` class to hold more UI-related values in its properties and avoid computing them in the UI layer. These changes are needed for the next PR: https://github.com/woocommerce/woocommerce-android/pull/9852

### Testing instructions

1. The order creation screen and items list should work as usual. No visual change should occur.
2. Unit tests should pass.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/MapItemToProductUiModel.kt [hunk 1]
```diff
@@ -31,10 +40,23 @@ class MapItemToProductUiModel @Inject constructor(
...
+                    discountAmount = decimalFormatter(item.discount),
+                    priceAfterDiscount = decimalFormatter(item.subtotal - item.discount)
                 )
-                // TODO check if we need to disable the plus button depending on stock quantity
             }
         }
     }
 }
+
+private fun getDecimalFormatter(
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/MapItemToProductUiModel.kt:55]: This is from `ProductItemView`, and `ProductItemView` will be removed eventually when this feature is complete.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/MapItemToProductUiModel.kt [hunk 2]
```diff
@@ -5,15 +5,19 @@ import com.woocommerce.android.ui.products.ProductDetailRepository
...
 
 class MapItemToProductUiModel @Inject constructor(
     private val dispatchers: CoroutineDispatchers,
     private val variationDetailRepository: VariationDetailRepository,
-    private val productDetailRepository: ProductDetailRepository
+    private val productDetailRepository: ProductDetailRepository,
+    private val currencyFormatter: CurrencyFormatter,
 ) {
-    suspend operator fun invoke(item: Order.Item): ProductUIModel {
+    suspend operator fun invoke(item: Order.Item, currencySymbol: String? = null): ProductUIModel {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/MapItemToProductUiModel.kt:19]: Maybe to add a test?

**Review by kidinov — APPROVED** [review]
LGTM. I left a suggestion to add a test on the mapper though

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/MapItemToProductUiModel.kt (hunk 2 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/MapItemToProductUiModel.kt:19]: Good idea. I'm going to add tests in a separate PR.

**Pull request merged**